### ISSUE TITLE
[Windows] Can't stop mongod process

### ISSUE THREAD

**Issue by dominic-p** [nodkz/mongodb-memory-server#193]
I'm running tests with mocha and I have the following code to stop the mongo server:

```ts
after(function (done) {
  this.timeout(10000);

  db.close()
    .then(() => mongoServer.stop())
    .then(() => { done(); })
    .catch((error) => { console.log('Error stopping mongo server', error); done(); });
  });
});
```

When I run the tests, I get an error message logged in the catch statement. I enabled debug mode and saw this output:

```
Called MongoMemoryServer.stop() method
Called MongoMemoryServer.ensureInstance() method:
Shutdown MongoDB server on port 55555 with pid 11724
Mongo[55555]: Called MongoInstance.kill():
Mongo[55555]:  - childProcess: send kill cmd...
Mongo[55555]:  - childProcess: got exit signal. Ok!
Mongo[55555]:  - killerProcess: send kill cmd...
Mongo[55555]: 2019-06-05T13:54:07.979-0700 I NETWORK  [conn1] end connection 127.0.0.1:55557 (3 connections now open)
2019-06-05T13:54:07.979-0700 I NETWORK  [conn2] end connection 127.0.0.1:55559 (2 connections now open)
2019-06-05T13:54:07.979-0700 I NETWORK  [conn3] end connection 127.0.0.1:55561 (1 connection now open)
2019-06-05T13:54:07.979-0700 I NETWORK  [conn4] end connection 127.0.0.1:55563 (0 connections now open)

Mongo[55555]: [MongoKiller]: exit - [null,"SIGTERM"]
Mongo[55555]:  - killerProcess: got exit signal. Ok!
Removing tmpDir C:\Users\UserName\AppData\Local\Temp\mongo-mem-16836ZeiScx7S058a
Error stopping mongo server { Error: EBUSY: resource busy or locked, unlink 'C:\Users\UserName\AppData\Local\Temp\mongo-mem-16836ZeiScx7S058a\mongod.lock'
    at Object.unlinkSync (fs.js:951:3)
    at _rmdirRecursiveSync (c:\path\to\project\node_modules\tmp\lib\tmp.js:309:12)
    at Object._cleanupCallback [as removeCallback] (c:\path\to\project\node_modules\tmp\lib\tmp.js:446:7)
    at MongoMemoryServer.<anonymous> (c:\path\to\project\node_modules\mongodb-memory-server-core\src\MongoMemoryServer.ts:162:14)
    at step (c:\path\to\project\node_modules\mongodb-memory-server-core\lib\MongoMemoryServer.js:43:23)
    at Object.next (c:\path\to\project\node_modules\mongodb-memory-server-core\lib\MongoMemoryServer.js:24:53)
    at fulfilled (c:\path\to\project\node_modules\mongodb-memory-server-core\lib\MongoMemoryServer.js:15:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Am I doing something wrong here?

**Comment by dominic-p** [comment]
My current workaround is to use [terminate](https://www.npmjs.com/package/terminate) to kill the child processes and then remove the temp dir manually. This seems to work, but it's definitely not pretty.

```
import { promisify } from 'util';
import { remove } from 'fs-extra';
import terminate from 'terminate';

const terminatePromise = promisify(terminate);

after(function () {

	this.timeout(10000);

	let instanceInfo;

	return db.close()
		.then(() => {

			instanceInfo = mongoServer.getInstanceInfo();

			return Promise.all([
				terminatePromise(instanceInfo.instance.childProcess.pid),
				terminatePromise(instanceInfo.instance.killerProcess.pid)
			]);
		})
		.then(() => remove(instanceInfo.dbPath))
		.catch((error) => {

			console.log('Error stopping mongo server', error);
		});
});
```

**Comment by hasezoey** [comment]
is this still a problem in newer versions?

**Comment by dominic-p** [comment]
@hasezoey, sorry, I don't use mongodb-memory-server anymore, so I can't say. The devs can feel free to close this issue if no one else is running into problems.

**Comment by hasezoey** [comment]
thanks for responding, i will close this, if anyone has this issue, feel free to comment / open a new issue

**Comment by lcrosetto** [comment]
I am still seeing this issue on Windows, but only when using the systemBinary.

**Comment by hasezoey** [comment]
@lcrosetto could you test this branch and report the results? https://github.com/hasezoey/mongodb-memory-server/tree/fix/testRemoveCrossSpawn

(cant test, dont have an windows machine)

**Comment by lcrosetto** [comment]
> @lcrosetto could you test this branch and report the results? https://github.com/hasezoey/mongodb-memory-server/tree/fix/testRemoveCrossSpawn
> 
> (cant test, dont have an windows machine)

It works for me, with a minor change in MongoBinary.ts:

https://github.com/hasezoey/mongodb-memory-server/commit/0806e9849298e1388ef2a0173bc6e42e71979508#commitcomment-43951745

**Comment by hasezoey** [comment]
@lcrosetto could you test again with the new commit (hasezoey@7d89e7c846d962d09d3676f5f73f0174bff86de2)?

**Comment by lcrosetto** [comment]
> @lcrosetto could you test again with the new commit ([hasezoey/mongodb-memory-server@7d89e7c](https://github.com/hasezoey/mongodb-memory-server/commit/7d89e7c846d962d09d3676f5f73f0174bff86de2))?

That works nicely. Thanks.

**Comment by hasezoey** [comment]
closing because fixed since `7.0.0-beta.11`
